File: src/main/java/br/com/blz/testjava/domain/model/Inventory.java
package br.com.blz.testjava.domain.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.Optional;

public class Inventory implements Serializable {

  private List<Warehouse> warehouses;

  public List<Warehouse> getWarehouses() {
    return warehouses;
  }

  public void setWarehouses(List<Warehouse> warehouses) {
    this.warehouses = warehouses;
  }

  public Integer getQuantity() {
    return Optional.ofNullable(warehouses)
      .orElse(new ArrayList<>())
      .stream().mapToInt(Warehouse::getQuantity).sum();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Inventory inventory = (Inventory) o;
    return Objects.equals(warehouses, inventory.warehouses);
  }

  @Override
  public int hashCode() {
    return Objects.hash(warehouses);
  }
}

File: src/main/kotlin/br/com/blz/testjava/domain/model/Inventory.kt
package br.com.blz.testjava.domain.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import org.hibernate.annotations.Parent
import javax.persistence.*

@Entity
open class Inventory (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    open var id: Long,

    @ManyToMany(cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    open var warehouses: MutableCollection<Warehouse>

) {

    /**
     * <p>Calcula o total de produtos em estoque.</p>
     *
     * @return total de produtos
     *
     * @see Warehouse#quantity
     */
    @JsonProperty("quantity")
    open fun calculateTotalQuantity() = warehouses.parallelStream().mapToInt(Warehouse::quantity).sum()

}